.vscode/launch.json

```diff
@@ -25,6 +25,24 @@
       "autoAttachChildProcesses": true,
       "sourceMaps": true,
       "preLaunchTask": "npm: build-vscode"
+    },
+    {
+      "type": "extensionHost",
+      "request": "launch",
+      "name": "Debug VSCode Tests",
```

Comment:
yes that is true, admittedly I didn't consider it and was just using my local instance, which may not be ideal in the long run

I'm playing with a few ideas to spin up a test container when we launch.. I've had _some_ success with creating a new task that spins one up.. the problem is tearing it down afterwards

 did you have any thoughts on how we could do this?

Reply:
okay, what do you reckon to this change?


https://github.com/neo4j/cypher-language-support/pull/232/commits/85beda8d5e584378b90fa94ab7fcfc31c30faa49

I was finding that with the preLaunchTask, creating a testcontainer and loading config with dotenv was working fine, but I think where this fell down was the dotenv variables were not loaded into the process that launches vscode.. I was finding that the container was also ending prematurely.. I wondered if this had something to do with the container shutting down when process/thread used to run the preLaunchTask was killed/returned to the worker pool.

in this commit, I'm creating a debug version of the testRunner entry point that creates and starts a testcontainer for the tests to use when debugging.. I don't know if this is the cleanest way of achieving this.. but it seems to work well enough and keeps the concerns of running tests in a debugging context and CI/CLI context separate

Reply:
Nice!

I think the only thing left to do is close the container? I don't know if that's possible or we'd need to clean up already running containers before we spin up a new one?

Reply:
yeah I was thinking about this

I'm observing that the container does seem to clean itself up once the tests finish and the instance of vscode closes.. like you said, I expect it might be sending a kill signal to child processes that's stopping the containers

it would be nice to explicitly close the container in code to avoid them sticking around

there must be a lifecycle hook we can use to do this

Reply:
Happy with the process killing all children as it seems to be happening. Good work!